src/IssueViz/Editor/QuickActions/QuickFixes/QuickFixActionsSource.cs

```diff
@@ -74,21 +74,28 @@ internal sealed class QuickFixActionsSource : ISuggestedActionsSource
         public event EventHandler<EventArgs> SuggestedActionsChanged;
 
         public IEnumerable<SuggestedActionSet> GetSuggestedActions(
-            ISuggestedActionCategorySet requestedActionCategories, 
+            ISuggestedActionCategorySet requestedActionCategories,
             SnapshotSpan range,
             CancellationToken cancellationToken)
         {
             var allActions = new List<ISuggestedAction>();
 
-            if (IsOnIssueWithApplicableQuickFixes(range, out var issuesWithFixes))
+            try
             {
-                foreach (var issueViz in issuesWithFixes)
+                if (IsOnIssueWithApplicableQuickFixes(range, out var issuesWithFixes))
                 {
-                    var applicableFixes = issueViz.QuickFixes.Where(x => x.CanBeApplied(textView.TextSnapshot));
+                    foreach (var issueViz in issuesWithFixes)
+                    {
+                        var applicableFixes = issueViz.QuickFixes.Where(x => x.CanBeApplied(textView.TextSnapshot));
 
-                    allActions.AddRange(applicableFixes.Select(fix => new QuickFixSuggestedAction(fix, textView.TextBuffer, issueViz, quickFixesTelemetryManager, logger)));
+                        allActions.AddRange(applicableFixes.Select(fix => new QuickFixSuggestedAction(fix, textView.TextBuffer, issueViz, quickFixesTelemetryManager, logger)));
+                    }
                 }
             }
+            catch (Exception ex) when (!ErrorHandler.IsCriticalException(ex))
+            {
+                logger.WriteLine(string.Format(Resources.ERR_QuickFixes_Exception, ex));
```

Comment:
I opted to log the entire call stack at normal verbosity. The issue is timing-related and hard to reproduce so we don't expect it to occur often.
If we log it at debug verbosity there's no chance we'll ever see it reported in the wild by users.